src/main/kotlin/io/github/rybalkinsd/kohttp/interceptors/SigningInterceptor.kt

```diff
@@ -0,0 +1,38 @@
+package io.github.rybalkinsd.kohttp.interceptors
+
+import okhttp3.HttpUrl
+import okhttp3.Interceptor
+import okhttp3.Response
+
+/**
+ * Request Signing Interceptor
+ *
+ * Enables signing of query parameters.
+ *
+ * @param parameterName the name of the parameter with signed key
+ * @param signer function with HttpUrl as a receiver to sign the request parameter
+ *
+ * @see HttpUrl
+ *
+ * @since 0.8.0
+ * @author gokul
+ */
+
+class SigningInterceptor(private val parameterName: String, private val signer: HttpUrl.() -> String) : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
```

Comment:
looks much better 👍 
also it's possible to replace` } return ` with `=`
```suggestion
    override fun intercept(chain: Interceptor.Chain): Response =
```